transports/p2p/addrmgr/addrmanager.go

```diff
@@ -146,6 +151,17 @@ func (a *AddrManager) updateAddress(netAddr, srcAddr *wire.NetAddress) {
 	}
 
 	addr := NetAddressKey(netAddr)
+
+	// If address in banned addresses and its bantime still active, ignore it
+	// otherwise, remove it from banned addresses and add it again
+	if t, ok := a.addrBanned[addr]; ok {
+		if t.After(time.Now()) {
+			a.Log.Debug().Msgf("New address %s still banned, ignoring", addr)
```

Comment:
Wouldn't that message appear too many times? If so, then maybe we can use the approach with [logging.SampledLogger](https://github.com/bitcoin-sv/block-headers-service/blob/abbc8b47902fc4b695f2be7111cf6161d42a5d2d/logging/logging.go#L73)

Reply:
Not really, it only happens when a peer sends us NewPeer message, which is actually very rarely.